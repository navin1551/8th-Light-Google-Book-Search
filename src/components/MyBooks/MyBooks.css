.my-book-area {
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  border: 1px solid black;
  list-style: none;
}

#my-book-images {
  margin-top: 10px;
}
