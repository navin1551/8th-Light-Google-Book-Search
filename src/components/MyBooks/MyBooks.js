import React from "react";
import "./MyBooks.css";

export default class MyBooks extends React.Component {
  render() {
    return (
      <div className="my-book-area">
        <li>
          <img
            src={this.props.image}
            id="my-book-images"
            alt="reading-list-images"
          />
          <p>Title: {this.props.title}</p>
          <p>Author: {this.props.author}</p>
          <p>Publisher: {this.props.publisher}</p>
        </li>
      </div>
    );
  }
}
