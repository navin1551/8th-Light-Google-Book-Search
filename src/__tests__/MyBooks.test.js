import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import MyBooks from "../components/MyBooks/MyBooks";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MyBooks />, div);
});

it("renders MyBooks correctly", () => {
  render(<MyBooks />);
});

it("matches snapshot", () => {
  const tree = renderer.create(<MyBooks />).toJSON();
  expect(tree).toMatchSnapshot();
});
