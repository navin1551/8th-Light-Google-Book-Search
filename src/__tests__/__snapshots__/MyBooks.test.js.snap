// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="my-book-area"
>
  <img
    alt="reading-list-images"
    id="my-book-images"
  />
  <p>
    Title: 
  </p>
  <p>
    Author: 
    N/A
  </p>
  <p>
    Publisher: 
    N/A
  </p>
</div>
`;
